src/comments/CommentItem.js

```diff
@@ -60,6 +60,11 @@ export default class CommentItem extends Component {
     }
   }
 
+  updateCommentPostion = () => {
+    // should happen after react-router updated the route
+    setTimeout(() => this.checkHashLink());
```

Comment:
Not a necessary hack. I think you can just use `componentDidMount` to call `checkHashLink()` and use `@withRouter` from `react-router` package to have access to `this.props.location` in case.